### PR TITLE
Added new readiness mode

### PR DESCRIPTION
**Pull request by skhomuti**
New oracle mode checks consensus, execution, and keys-api services to ensure they are ready to work with the oracle daemon.
All checks will be checked on two block stamps - last finalized and frame ago from it.

image

Additional little change here - [Add endpoint to the error text](https://github.com/lidofinance/lido-oracle/pull/301/commits/aa1c950b7ad1c82936dda8e1500774af4d0d8b88)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,11 +34,17 @@ docker build -t lidofinance/oracle .
 \
```
Comment (F4ever) [inline README.md:34]: How to pass variables

**Comment by F4ever** [comment]
Do this checks without lido contracts.
Main things we want to check is that Node's and Keys API service setuped correctly.

File: src/main.py [hunk 2]
```diff
@@ -69,12 +69,15 @@ def main(module_name: OracleModule):
...
 
     web3.attach_modules({
-        'lido_contracts': LidoContracts,
         'lido_validators': LidoValidatorsProvider,
         'transaction': TransactionUtils,
         'cc': lambda: cc,  # type: ignore[dict-item]
         'kac': lambda: kac,  # type: ignore[dict-item]
     })
+    if variables.LIDO_LOCATOR_ADDRESS:
+        web3.attach_modules({
```
Comment (F4ever) [inline src/main.py:78]: there could be some exception if no LIDO_LOCATOR_ADDRESS were setup and we run Accounting module.
Maybe create another entrypoint for this one?

File: src/main.py [hunk 3]
```diff
@@ -83,31 +86,31 @@ def main(module_name: OracleModule):
...
+def check_required_variables(module_name: OracleModule):
     errors = []
     if '' in variables.EXECUTION_CLIENT_URI:
         errors.append('EXECUTION_CLIENT_URI')
     if variables.CONSENSUS_CLIENT_URI == '':
         errors.append('CONSENSUS_CLIENT_URI')
     if variables.KEYS_API_URI == '':
         errors.append('KEYS_API_URI')
-    if variables.LIDO_LOCATOR_ADDRESS in (None, ''):
+    if variables.LIDO_LOCATOR_ADDRESS in (None, '') and module_name != OracleModule.READINESS:
```
Comment (F4ever) [inline src/main.py:113]: Maybe create separate flow for node testing?

File: README.md (hunk 1 above)
Comment (skhomuti) [reply README.md]: added

File: src/main.py (hunk 3 above)
Comment (skhomuti) [reply src/main.py]: fixed, take a look at

File: src/modules/readiness/checks/common.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+
+def check_accounting_contract_configs(accounting):
```
Comment (F4ever) [inline src/modules/readiness/checks/common.py:2]: If LIDO_LOCATOR provided, run this tests

File: src/main.py [hunk 5]
```diff
@@ -82,30 +83,40 @@ def main(module_name: OracleModule):
...
         accounting = Accounting(web3)
         accounting.check_contract_configs()
         accounting.run_as_daemon()
-    elif module_name == OracleModule.EJECTOR:
+    elif module == OracleModule.EJECTOR:
         logger.info({'msg': 'Initialize Ejector module.'})
         ejector = Ejector(web3)
         ejector.check_contract_configs()
         ejector.run_as_daemon()
+    elif module_name == OracleModule.READINESS:
```
Comment (F4ever) [inline src/main.py:96]: READINESS -> to test

Comment (skhomuti) [reply src/main.py]: fixed

Comment (skhomuti) [reply src/main.py]: image

**Review by F4ever — APPROVED** [review]

**Pull request merged**